scheduler/cloudformation/cloudformation.go

```diff
@@ -274,9 +277,138 @@ func (s *Scheduler) submit(ctx context.Context, tx *sql.Tx, app *scheduler.App,
 		}
 	}
 
+	go s.waitForDeploymentToStabilize(ctx, app, stackName, events, done)
```

Comment:
Since this method does things like describing the stack, it should only be operating while the stack operation lock is held, otherwise multiple concurrent `Submits` will produce the wrong results.